src/transformStateWithClones.js

```diff
@@ -8,6 +8,49 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const k = [];
+  let stateClone = {
+    ...state,
+  };
+
+  for (const action in actions) {
+    const { type } = actions[action];
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = actions[action];
+
+        stateClone = {
+          ...stateClone,
+          ...extraData,
+        };
+        k.push({ ...stateClone });
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = actions[action];
+
+        for (const key of keysToRemove) {
+          for (const kInStat in stateClone) {
+            if (key === kInStat) {
+              delete stateClone[key];
+            }
+          }
+        }
+        k.push({ ...stateClone });
+        break;
+
+      case 'clear':
+        stateClone = {};
+        k.push({ ...stateClone });
```

Comment:
You push it for each case. Just do it after all cases
![image](https://user-images.githubusercontent.com/96409563/227859003-b25c3a47-04aa-4fb2-a835-a6ea7699d1a4.png)